st2common/st2common/models/api/keyvalue.py

```diff
@@ -114,17 +120,22 @@ def from_model(cls, model, mask_secrets=True):
         if 'id' in doc:
             del doc['id']
 
-        if model.expire_timestamp:
+        if getattr(model, 'expire_timestamp', None) and model.expire_timestamp:
             doc['expire_timestamp'] = isotime.format(model.expire_timestamp, offset=False)
 
         encrypted = False
-        if model.secret:
+        secret = getattr(model, 'secret', False)
+        if secret:
             encrypted = True
 
-        if not mask_secrets and model.secret:
+        if not mask_secrets and secret:
             doc['value'] = symmetric_decrypt(KeyValuePairAPI.crypto_key, model.value)
             encrypted = False
 
+        scope = getattr(model, 'scope', SYSTEM_KV_PREFIX)
```

Comment:
I personally view scope and key prefix as two different things and they don't need to be the same (scope is a user visible feature and key prefix for a particular scope is an implementation detail).

In any case, I would have those scopes:

* `user` - User scopes, keys are prefixed with some common prefix for this scope (`user`.<user pk aka username>.foo` or `_user.<user pk aka username>.foo`)
* `global` - Anything which doesn't have a prefix (so any other existing key). We could explicitly prefix keys with a global scope, but the question is how to handle the keys without the prefix then.

Or do you want to introduce also a concept of no scope?